pages/docs/tutorials/native/apple-framework.md

```diff
@@ -0,0 +1,284 @@
+```
+</div>
+
+The name `DemoDemoClazz` and `DemoDemoObject` are created for `DemoClazz` and `DemoObject` 
+Kotlin classes. The `Demo` prefix comes from the `-output` parameter
+of the `kotlinc` compiler and the Framework name. All global functions from Kotlin
+are turned into `Demo` class in Objective-C/Swift, where `Demo` is the Framework name again.
+You may notice, that Kotlin `String` and Objective-C `NSString` are mapped transparently.
+Similarly, `Unit` type from Kotlin is mapped to `void`. We see primitive types
+are mapped directly.
+
+You may want to see [Objective-C Interop](https://github.com/JetBrains/kotlin-native/blob/master/OBJC_INTEROP.md)
+documentation article to learn all other cases in detail.
+
+## Using the Code from Objective-C
+
+Let's call the Kotlin Framework from Objective-C. For that we create the `main.m` file with 
+the following content:
+
+<div class="sample" markdown="1" mode="obj-c" theme="idea" data-highlight-only="1" auto-indent="false">
+
+```obj-c 
+
+#import <Foundation/Foundation.h>
+#import <Demo/Demo.h>
+
+int main(int argc, const char * argv[]) {
+    
+    [[DemoDemoObject demoObject] field];
+    
+    DemoDemoClazz* clazz = [[ DemoDemoClazz alloc] init];
+    [clazz memberFunctionP:42];
+    
+    [Demo forIntegersB:1 s:1 i:3 l:4];
+   
+    NSString* ret = [Demo stringsStr:(@"That is string")];
+    return 0;
+}
+```
+</div>
+
+We call Kotlin classes directly from Objective-C code. Kotlin `object` has the class method 
+function `demoObject`, which allows us to get the only instance of the object and to call 
+`DemoObject` methods on it. 
+The standard pattern is used to create an instance of the `DemoClazz` class. We call
+the `[[ DemoDemoClazz alloc] init]` on Objective-C.
+Global functions from the Kotlin sources are scoped under the `Demo` class on Objective-C.
+All methods are turned into class methods of that class.
+The `strings` function is turned into `Demo.stringsStr` function in Objective-C, we are
+able to pass `NSString` directly to it. The return is visible as `NSString` too.
+
+## Using the Code from Swift
+
+The Framework, that we compiled with Kotlin/Native, has helper attributes to make it
+easier to with it from Swift. Let's convert the previous Objective-C example
+into Swift. As the result we'll have the following code in `main.swift`:
+
+<div class="sample" markdown="1" mode="swift" theme="idea" data-highlight-only="1" auto-indent="false">
+
+```swift
+let field = DemoDemoObject().field
+
+let clazz = DemoDemoClazz()
+clazz.memberFunction(p: 42)
+
+Demo.forIntegers(b: 1, s: 2, i: 3, l: 4)
+
+let ret = Demo.strings(str: "That is string")
+``` 
+</div>
+
+The Kotlin code is turned into nearly the similar looking
+code in Swift. There are some small differences, so far. We create the Kotlin `object DemoObject`
+as a class in Swift to access it's methods. We may see from the Objective-C attributes 
+in `<Framework>/Headers/Demo.h` that it is the way to access the object. 
```

Comment:
It is not exactly clear from this sentence that `DemoObject()` doesn't create new instance of `DemoObject` but returns the existing single one. I would prefer to avoid the word "create".

Reply:
fixed